### ISSUE TITLE
Serialize.Linq and .Net Core 2.2

### ISSUE THREAD

**Issue by Michael-Kalo** [esskar/Serialize.Linq#111]
Hello,

I have evaluated Serialize.Linq in a "normal" .Net 4.7.2 project, which uses WCF services and SOAP and it works great. No problems at all. Everything as expected.

Now I am trying to use Serialize.Linq in a .Net Core 2.2 project to receive expressions.

To do that, I followed your Client/Server examples from here:

Client:
https://github.com/esskar/Serialize.Linq/blob/master/src/Serialize.Linq.Examples/Serialize.Linq.Examples.RestClient/Program.cs

Server:

https://github.com/esskar/Serialize.Linq/blob/master/src/Serialize.Linq.Examples/Serialize.Linq.Examples.RestApi/Controllers/PersonController.cs

The problem:

On the server side, the PUT request arrives, but the ExpressionNode is always Null. So it seems, it doesn't get deserialized from JSON. 

If I use that same PUT request to send any other complex object, it will arrive serialized at the server. (Not Null)

Inside the .Net Core 2.2 project, I tried different versions (references) of Serialize-Linq. The Net 4.5, the NetStandard 1.3 and NetStandard 2.0 versions.

My question is: Have you already successfully tested Serialize.Linq with .Net Core 2.2 ?

Thank you!

**Comment by esskar** [comment]
Hi Michael, let me update the examples over the weekend.

**Comment by Michael-Kalo** [comment]
Hi Sascha, thanks for your reply! 

That would be great. My guess is, that we have to plug in a more or less specialized, or at least configured, JSON Serializer into the ASP.Net Core pipeline. (OutputFormatters | InputFormatters)

When I look into HttpContext.Request, I can see, that the ExpressionNode object is in the body of the request. So it is there, but no deserialization.

Something like this: (Pseudo-Code)

`
services.AddMvc(options =>
{
           options.InputFormatters.Add(new ExpressionNodeSerializer());

 }).SetCompatibilityVersion(Microsoft.AspNetCore.Mvc.CompatibilityVersion.Version_2_2);

`
....but let's see.....

**Comment by Michael-Kalo** [comment]
Hello Sascha,

I have tested my idea from above with XML serializers and that works.

My client looks like this:

ExpressionNode queryNode = query.ToExpressionNode();
var xmlFormatter = new XmlMediaTypeFormatter();
var response = await client.PostAsync(ai.ApiTemplate, queryNode, xmlFormatter, "application/xml", cancellationToken);

On the server side, I plugged in XML input/output formatters. It looks like this:

 services.AddMvc(options =>
            {
                options.RespectBrowserAcceptHeader = true;
                options.ReturnHttpNotAcceptable = true;
                options.OutputFormatters.Add(new XmlDataContractSerializerOutputFormatter());
                options.InputFormatters.Add(new XmlDataContractSerializerInputFormatter());

            }).SetCompatibilityVersion(Microsoft.AspNetCore.Mvc.CompatibilityVersion.Version_2_2);

That works, the ExpressionNode arrives deserialized in the controller. (Not Null)

=====

From your examples, you propose to use the client formatter like this:

 JsonMediaTypeFormatter formatter = new JsonMediaTypeFormatter();
                formatter.SerializerSettings = new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.Objects };

May it be the solution, that the default JSON InputFormatter, that is used by .Net Core on the server side needs this JsonSerializerSettings?


Could it be, that we have to change the JsonSerializerSettings inside JsonInputFormatter, because on the client side
